docs/source/development/contributing/overview.md

```diff
@@ -0,0 +1,340 @@
+environment and check it is all working.
+
+```sh
+poetry run pre-commit install
+poetry run pre-commit run --all-files
+```
+
+That might take a little while to run on the first use. Once you have done this, every
+`git commit` will generate similar output and your commit will fail if issues are found.
+
+### Static typing with `mypy`
+
+The `python` programming language does not _require_ code objects to be typed, but the
+`pyrealm` package uses [type hints](https://peps.python.org/pep-0484/) to annotate code.
+Those type hints are then checked using the `mypy` static type checker, which is
+installed by `poetry` and is run as one of the `pre-commit` checks.
+
+The `mypy` package and the plugins we use are all installed by `poetry`. See the [code
+quality assurance page](./code_qa_and_typing.md) for more information on using `mypy`.
+
+### Package testing
+
+All code in the `pyrealm` package should have accompanying unit tests, using `pytest`.
+Look at the existing test suite in the `tests/unit` directory to see the structure and
+get a feel for what they should do, but essentially unit tests should provide a set of
+known inputs to a function and check that the expected answer (which could be an
+Exception) is generated.
+
+Again, the `pytest` package and plugins are installed by `poetry`. See the [code testing
+page](./code_testing.md) for more details but you should be able to check the tests run
+using the following command. Be warned that the `mypy` steps can be very time consuming
+on the first run, but `pytest` does some cacheing that makes them quicker when they next
+run.
+
+```sh
+poetry run pytest
+```
+
+### The `example_data` module
+
+The `virtual_ecosystem` package includes the `example_data` submodule TODO: LINK
+
+### Documentation
+
+We use `sphinx` to maintain the documentation for `pyrealm` and Google style docstrings
+using the `napoleon` formatting to provide API documentation for the code. We use MyST
+Markdown to provide dynamically built usage examples. See the [documentation
+pages](../documentation/documentation.md) for details but to get started, the following
+code can be used to build the documentation.
+
+```bash
+# Build docs using sphinx
+cd docs
+poetry run sphinx-build -W --keep-going source build
+```
+
+Once that command completes, the file `docs/build/html/index.html` can be opened to view
+the built documentation.
+
+### GitHub Actions
+
+We use GitHub Action workflows to update `pre-commit`, run code quality checks on pull
+requests, and to automate profiling and release publication. See the [GitHub Actions
+page](./github_actions.md) for details.
+
+### Package version releases
+
+We use trusted publishing from GitHub releases to release new versions of `pyrealm` to
+[PyPI](https://pypi.org/project/pyrealm/). Releases are also picked up and archived on
+[Zenodo](https://doi.org/10.5281/zenodo.8366847). See the [release process
+page](./release_process.md) for details.
```

Comment:
??

Reply:
We don't do this on Virtual Ecosystem. Truth be told, I'm not sure why we do it on `pyrealm`.